game_api.txt

```diff
@@ -473,18 +473,18 @@ The player API can register player models and update the player's appearance.
 ### Model Definition
 
 	{
-		animation_speed = 30,           -- Default animation speed, in FPS
-		textures = {"character.png", }, -- Default array of textures
-		visual_size = {x = 1, y = 1},   -- Used to scale the model
+		animation_speed = 30,           -- Default animation speed, in keyframes per second
+		textures = {"character.png"},   -- Default array of textures
 		animations = {
-			-- <anim_name> = {x = <start_frame>, y = <end_frame>},
-			foo = {x = 0, y = 19},
-			bar = {x = 20, y = 39},
-			-- ...
+			-- [anim_name] = {x = <start_frame>, y = <end_frame>, collisionbox = default collisionbox, eye_height = default eye height},
+			stand = ..., lay = ..., walk = ..., mine = ..., walk_mine = ..., -- required animations
+			sit = ... -- used by boats and other MTG mods
 		},
-		collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3}, -- In nodes from feet position
-		stepheight = 0.6,                                -- In nodes
-		eye_height = 1.47,                               -- In nodes above feet position
+		-- Default object properties, see lua_api.txt
+		visual_size = {x = 1, y = 1},
+		collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3},
+		stepheight = 0.6,
+		eye_height = 1.47
```

Comment:
Why are single-value collisionbox and eye_height still required in the def when the new approach is to define them in the animations?
Seems messy and confusing, i think these should only be defined in animations.

Reply:
These are default values if there are no `eye_height` or `collisionbox` values provided by animations. This allows for increased code quality (no repetition of these values) and backwards compatibility.